patches/kaggle_gcp.py

```diff
@@ -200,9 +200,70 @@ def monkeypatch_gcs(self, *args, **kwargs):
         storage.Client.__init__ = monkeypatch_gcs
     return storage
 
+def init_automl():
+    is_user_secrets_token_set = "KAGGLE_USER_SECRETS_TOKEN" in os.environ
+    from google.cloud import automl_v1beta1 as automl
+    if not is_user_secrets_token_set:
+        return automl
+
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_automl():
+        return automl
+
+    from kaggle_secrets import GcpTarget
+    from kaggle_gcp import KaggleKernelCredentials
+    kaggle_kernel_credentials = KaggleKernelCredentials(target=GcpTarget.AUTOML)
+
+    # The AutoML client library exposes 4 different client classes (AutoMlClient,
+    # TablesClient, PredictionServiceClient and GcsClient), so patch each of them.
+    # The same KaggleKernelCredentials are passed to all of them.
+
+    automl_client_init = automl.AutoMlClient.__init__
+    def monkeypatch_automl(self, *args, **kwargs):
+        specified_credentials = kwargs.get('credentials')
+        if specified_credentials is None:
+            Log.info("No credentials specified, using KaggleKernelCredentials.")
+            kwargs['credentials'] = kaggle_kernel_credentials
+        # Note: This is only here so that unit tests can check whether
+        # credentials were set properly.
+        self._kaggle_credentials = kwargs['credentials']
```

Comment:
I don't like this, but it's the only way that I've been able to verify that the patch actually runs and does something.

Reply:
Another possible way to test is to mock patch the transport (https://github.com/googleapis/google-cloud-python/blob/master/automl/google/cloud/automl_v1beta1/gapic/auto_ml_client.py#L264) and verify that the credentials passed in are the ones you're expecting, sort of like in https://github.com/Kaggle/docker-python/blob/master/tests/test_bigquery.py#L108

Reply:
Done. Thanks for the tip! As it turns out I had to do something similar in my [PR](https://github.com/googleapis/google-cloud-python/pull/9299) for the AutoML client libraries, so I've spent a lot of time with `patch` these last few days.